src/libraries/Microsoft.PowerFx.Connectors/Shared/ServiceFunction.cs

```diff
@@ -285,6 +287,78 @@ public override bool CheckForDynamicReturnType(TexlBinding binding, TexlNode[] a
         }
 #endif
 
+        public override async Task<List<string>> GetConnectorSuggestionsAsync(CallNode callNode, int argPosition, CancellationToken cts)
+        {
+            if (argPosition >= 0 && argPosition < _requiredParameters.Length)
+            {
+                ConnectorDynamicValue cdv = _requiredParameters[argPosition].ConnectorDynamicValue;
+
+                if (cdv == null || cdv.ServiceFunction == null)
+                {
+                    return null;
+                }
+
+                List<FormulaValue> arguments = new List<FormulaValue>();
+
+                foreach (ServiceFunctionParameterTemplate sfpt in cdv.ServiceFunction._requiredParameters)
+                {
+                    string paramName = sfpt.TypedName.Name;
+                    DType paramType = sfpt.TypedName.Type;
+
+                    string currentFunctionParamName = cdv.ParameterMap.FirstOrDefault(kvp => kvp.Value == paramName).Value;
+                    int currentFunctionParamIndex = _requiredParameters.FindIndex(st => st.TypedName.Name == currentFunctionParamName);
+                    TexlNode texlNode = callNode.Args.ChildNodes[currentFunctionParamIndex];
+
+                    FormulaValue arg = texlNode switch
+                    {
+                        StrLitNode str => FormulaValue.New(str.Value),
+                        NumLitNode num => FormulaValue.New(num.ActualNumValue),
+                        _ => null
+                    };
+
+                    if (arg == null)
+                    {
+                        return null;
+                    }
+
+                    arguments.Add(arg);
+                }
+
+                cts.ThrowIfCancellationRequested();
+                FormulaValue result = await cdv.ServiceFunction.InvokeAsync(arguments.ToArray(), cts);
+
+                List<string> suggestions = new List<string>();
+                if (result is RecordValue rv)
+                {
+                    if (!string.IsNullOrEmpty(cdv.ValueCollection) && !string.IsNullOrEmpty(cdv.ValuePath))
```

Comment:
nit - can we check this earlier? Before the network call?

Or even at cdv creation time?

Reply:
cdv creation time -> those are optional params so that's not the right place
before the call -> we _could_ but I imagine we could add support for more combinations later - the the cases that not supported today & that would duplicate efforts